### PR TITLE
Update app template

### PR DESCRIPTION
**Pull request by Aschen**
## What does this PR do?

Update the application template:
 - bump dependencies
 - add github actions
 - add functional test framework

### CODE REVIEW

**Review by scottinet — APPROVED** [review]

File: templates/app-scaffold/ts/.github/workflows/pull_request.workflow.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+name: Pull request checks
```
Comment (xbill82) [inline templates/app-scaffold/ts/.github/workflows/pull_request.workflow.yml:1]: is it a good idea to scaffold the github actions? Just wondering.

**Review by xbill82 — APPROVED** [review]

File: templates/app-scaffold/ts/.github/workflows/pull_request.workflow.yml (hunk 1 above)
Comment (Aschen) [reply templates/app-scaffold/ts/.github/workflows/pull_request.workflow.yml:1]: IMHO Github actions are part of the standard Kuzzle workflow which includes a linter and functional tests (like Vue.js/Webpack does if I remember well)

If they don't want to use it they can just delete the files but IMHO users shouldn't

Comment (alexandrebouthinon) [reply templates/app-scaffold/ts/.github/workflows/pull_request.workflow.yml:1]: I partly agree with Luca. Maybe we should however remove the deployment part which can be very different depending on the use case

File: templates/app-scaffold/ts/.github/workflows/push_production.workflow.yml [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+            registry: registry.gitlab.com
+            username: ${{ secrets.GITLAB_DOCKER_USERNAME }}
+            password: ${{ secrets.GITLAB_DOCKER_TOKEN }}
+        - name: Build and push image
+          uses: docker/build-push-action@v2
+          with:
+            context: ./backend
+            file: ./backend/Dockerfile
+            push: true
+            tags: registry.gitlab.com/kuzzle/<CHANGE-ME>:latest
```
Comment (Leodau) [inline templates/app-scaffold/ts/.github/workflows/push_production.workflow.yml:72]: Is there a place where we give instructions to come here and update these **<CHANGE-ME>** ?

Comment (Aschen) [reply templates/app-scaffold/ts/.github/workflows/push_production.workflow.yml:72]: No, but as we agreed, the deployment part is now removed

**Pull request merged**